### PR TITLE
Task page

### PR DESCRIPTION
**Pull request by saketkum11**
- created task page 
-  task modal form

### CODE REVIEW

File: src/Component/ModalTask/ModalTask.jsx [hunk 1]
```diff
@@ -13,9 +38,25 @@ const ModalTask = ({ setModalTaskFlag }) => {
...
+                <form
+                  onSubmit={(event) => {
+                    event.preventDefault();
+                    taskDispatch({
+                      type: ACTION_TYPE.SUMBIT_TASK,
+                      payload: taskDetails,
+                    });
+                    notify(message);
+                    setModalTaskFlag((flag) => !flag);
+                  }}
```
Comment (MiheerTamkhane) [inline src/Component/ModalTask/ModalTask.jsx:57]: You can extract this function for more clean code. Suggestion!

File: src/Component/Tasks/Tasks.jsx [hunk 2]
```diff
@@ -1,4 +1,21 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useTask } from "../../Context/Task-context/Task-context";
```
Comment (MiheerTamkhane) [inline src/Component/Tasks/Tasks.jsx]: I think here is a typo?

File: src/Component/Tasks/Tasks.jsx [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+  const { taskState } = useTask();
+  const { task, tag } = taskState;
+
+  const navigate = useNavigate();
+
+  const taskHandler = (task) => {
+    navigate("/");
+  };
+
+  useEffect(() => {}, []);
```
Comment (MiheerTamkhane) [inline src/Component/Tasks/Tasks.jsx]: Empty useEffect?

File: src/Component/Tasks/Tasks.jsx [hunk 4]
```diff
@@ -7,112 +24,46 @@ const Tasks = () => {
...
-          <button className="mr-3 bg-gray-900 text-white w-20 h-7 rounded-2xl my-3">
-            work
-          </button>
+          {tag.map((tags) => {
+            return (
+              <button
+                key={tags}
+                className="mr-3 bg-gray-900 text-white px-3 py-1 rounded-2xl my-3"
+              >
+                {tags}
```
Comment (MiheerTamkhane) [inline src/Component/Tasks/Tasks.jsx]: I think the naming conventions should be `tags` on you are mapping and `tag` in the callback.

File: src/Component/Tasks/Tasks.jsx [hunk 5]
```diff
@@ -7,112 +24,46 @@ const Tasks = () => {
...
-            <div>
-              <button>
-                <i className="fa-solid fa-pen-to-square"></i>
-              </button>
-              <button>
-                <i className="fa-solid fa-trash-can"></i>
-              </button>
-            </div>
-          </div>
+          {task.map(({ title, _id }) => {
```
Comment (MiheerTamkhane) [inline src/Component/Tasks/Tasks.jsx]: again `tasks`.

File: src/Context/Task-context/Task-context.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { createContext, useContext, useReducer } from "react";
+import { taskReducer } from "../../Pages";
+
+const TaskContext = createContext();
+const useTask = () => useContext(TaskContext);
+
+const TaskProvider = ({ children }) => {
+  const initialState = {
+    task: [],
+    tag: ["Read Book", "Homework", "Drawing"],
```
Comment (MiheerTamkhane) [inline src/Context/Task-context/Task-context.js]: here in initialState make it `tasks` and `tags`.

File: src/index.js [hunk 7]
```diff
@@ -4,13 +4,16 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
+import { TaskProvider } from "./Context/Task-context/Task-context";
```
Comment (MiheerTamkhane) [inline src/index.js]: Here also Task-context typo

**Review by MiheerTamkhane — APPROVED** [review]
Everything is fine and working smoothly. 
Code structure is nice and code is also readable.
Make suggested changes and you are good to go. Great Work!

File: src/Component/ModalTask/ModalTask.jsx [hunk 8]
```diff
@@ -1,4 +1,29 @@
...
+  const { taskDispatch } = useTask();
+  const [taskDetails, setTaskDetails] = useState({
+    _id: uuid(),
+    title: "",
+    tag: "",
+    description: "",
+    time: "",
+  });
+
+  console.log("taskDestail from modalTask componet", taskDetails);
```
Comment (AnjaliDeshwani) [inline src/Component/ModalTask/ModalTask.jsx]: Remove logs

File: src/Component/ModalTask/ModalTask.jsx [hunk 9]
```diff
@@ -49,6 +102,12 @@ const ModalTask = ({ setModalTaskFlag }) => {
                   </div>
                   <div className="my-2">
                     <input
+                      onChange={(e) => {
+                        setTaskDetails({
+                          ...taskDetails,
+                          time: e.target.value,
+                        });
+                      }}
```
Comment (AnjaliDeshwani) [inline src/Component/ModalTask/ModalTask.jsx:107]: You can extract out  all the common task change handler in one function, like below-

 const taskChangeHandler = (e) => {
    setTaskDetails({ ...taskDetails, [e.target.name]: e.target.value });
  };

**Review by AnjaliDeshwani — APPROVED** [review]
Clean and readable code, changes look good to merge.
UI looks great.

**Review by shubhamkumar648 — APPROVED** [review]
- suggestion is already mentioned 
- remaining all perfect

**Review by jeetsdev — APPROVED** [review]
Suggestions are already mentioned,
Else everything looks good
UI is clean (Change the text color of the reset button)

File: src/Component/ModalTask/ModalTask.jsx (hunk 1 above)
Comment (saketkum11) [reply src/Component/ModalTask/ModalTask.jsx:57]: sure .

File: src/Component/Tasks/Tasks.jsx (hunk 2 above)
Comment (saketkum11) [reply src/Component/Tasks/Tasks.jsx]: sure will check

File: src/Component/Tasks/Tasks.jsx (hunk 3 above)
Comment (saketkum11) [reply src/Component/Tasks/Tasks.jsx]: removed it.

File: src/Component/Tasks/Tasks.jsx (hunk 4 above)
Comment (saketkum11) [reply src/Component/Tasks/Tasks.jsx]: yes modified it.

File: src/Context/Task-context/Task-context.js (hunk 6 above)
Comment (saketkum11) [reply src/Context/Task-context/Task-context.js]: yes changed it.

File: src/Component/Tasks/Tasks.jsx (hunk 5 above)
Comment (saketkum11) [reply src/Component/Tasks/Tasks.jsx]: yes modified

File: src/index.js (hunk 7 above)
Comment (saketkum11) [reply src/index.js]: Yes changed it.

File: src/Component/ModalTask/ModalTask.jsx (hunk 9 above)
Comment (saketkum11) [reply src/Component/ModalTask/ModalTask.jsx:107]: sure new thing .

File: src/Component/ModalTask/ModalTask.jsx (hunk 8 above)
Comment (saketkum11) [reply src/Component/ModalTask/ModalTask.jsx]: done

**Pull request merged**